src/components/SnackChat/SnackChat.js

```diff
@@ -97,15 +97,63 @@ class SnackChat extends Component {
 
     const body = {
       ears: calcAngles({
-        left: pose.keypoints[3].position,
-        right: pose.keypoints[4].position
+        leftX: normalise(pose.keypoints[3].position.x),
+        leftY: normalise(pose.keypoints[3].position.y),
+        rightX: normalise(pose.keypoints[4].position.x),
+        rightY: normalise(pose.keypoints[4].position.y)
       }),
       shoulders: calcAngles({
-        left: pose.keypoints[5].position,
-        right: pose.keypoints[6].position
+        leftX: normalise(pose.keypoints[5].position.x),
+        leftY: normalise(pose.keypoints[5].position.y),
+        rightX: normalise(pose.keypoints[6].position.x),
+        rightY: normalise(pose.keypoints[6].position.y)
       })
     };
 
+    this.i = ++this.i % POSITION_BUFFER_SIZE;
+    const forEachAttribute = callback =>
+      ['ears', 'shoulders'].forEach(bodyPart =>
+        [
+          'leftX',
+          'rightX',
+          'leftY',
+          'rightY',
+          'width',
+          'height',
+          'span',
+          'angle'
+        ].forEach(attribute => callback(bodyPart, attribute))
+      );
+
+    // position buffer will contain undefined during first iteration
+    if (this.positionBuffer.includes(undefined)) {
```

Comment:
is there anyway to not call this function the first time then we can remove this?
if its too hard dont worry

Reply:
This is called multiple times until the array of position data is full - 10 times currently which is a not-insignificant proportion of the total number of iterations. I think it is best to leave it as is